src/containers/Dashboard.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { Chart } from 'primereact/chart';
+import '@binaris/shift-babel-macro/macro';
+
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Container from 'react-bootstrap/Container';
+import Button from 'react-bootstrap/Button';
+
+import {
+  generateRemoteData,
+  setStorageData,
+  getStorageData,
+} from '../../backend/dashboard-server';
+import { Display } from '../constants/enums';
+
+export default class Dashboard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      chart_data: {
+        labels: ['A', 'B', 'C'],
+        datasets: [
+          {
+            label: 'My First dataset',
+            data: [100, 200, 300],
+            backgroundColor: ['#FF6384', '#13bdd8', '#FFCE56'],
+            hoverBackgroundColor: ['#FF6384', '#13bdd8', '#FFCE56'],
+          },
+        ],
+      },
+      display: Display.FRONTEND,
+    };
+    this.generateLocalData = this.generateLocalData.bind(this);
+    this.getRemoteData = this.getRemoteData.bind(this);
+    this.fetchFromStorageData = this.fetchFromStorageData.bind(this);
+    this.initStorageData();
+  }
+
+  initStorageData() {
+    setStorageData(this.state.chart_data.datasets[0].data);
```

Comment:
Can we initialize to something different from the local data so it's clear that something changed when fetching from remote?

Reply:
We ignore the possibility of an error here, we should catch the error and report it to the user, e.g: https://github.com/binaris/shiftjs/blob/262a886958214313f1bbed815d843b1513c6cb18/app-testsuite/app/src/App.js#L5